.github/workflows/ci-pipeline.yml

```diff
@@ -0,0 +1,22 @@
+name: Continuous Integration
+on: 
+  push: 
+    paths-ignore:
+      - 'README.md'
+      - '.dockerignore'
+      - 'my-ansible-*.yml'
+  pull_request:
+    paths-ignore:
+      - 'README.md'
+      - '.dockerignore'

```

Comment:
You've flagged this file to be ignored for your tests which makes sense, except I can't see a `.dockerignore` file checked in at all!

Given that you're fairly careful about what you check in that's not critical since we shouldn't be leaking any secrets, but I'd encourage adding one anyway to try and pre-empt any changes that might later accidentally leak secrets to our images, and also to prevent any caching that might cause issues, e.g. include both `.env` and `**/__pycache__` in your ignore file, plus various other irrelevant files!